### PR TITLE
[14.0][FIX]company_dependent_flag: fixed missing label

### PR DESCRIPTION
**Pull request by toita86**
Before this fix in the pop-up window for creating contacts, the phone field was missing the label.
By not updating at every iteration the existing labels its possible to have more than two element with the same label have the flag

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
LGTM

image

File: company_dependent_flag/models/base.py [hunk 1]
```diff
@@ -26,30 +26,28 @@ def _update_company_dependent_css(self, arch):
...
 
-                    # Create a new span element
-                    span = arch.createElement("span")
-                    span.setAttribute(
-                        "class",
-                        "fa fa-lg fa-building-o",
-                    )
-                    span.setAttribute("title", "Values set here are company-specific.")
+                    if parent_form.tagName != "form":
+                        continue  # Skip this field as it's in a subview
```
Comment (HekkiMelody) [inline company_dependent_flag/models/base.py]: suggestion(non-blocking): this could be cleaner and more elegant if it was in its own method, IMO.

**Review by HekkiMelody — APPROVED** [review]
Code review, LGTM.

Another non-blocking suggestion is to add a note in `Known Issues` that this won't work cleanly if the field in question is present twice in the view.  It's very rare, but it could happen and I think it's worth mentioning.

**Review by HekkiMelody — APPROVED** [review]
Code review, LGTM

**Comment by francesco-ooops** [comment]
@pedrobaeza what do you say?

@bealdav as module author too :)

File: company_dependent_flag/README.rst [hunk 2]
```diff
@@ -39,6 +39,12 @@ With this module all company dependent fields are decorated with an adhoc class.
 .. contents::
    :local:
 
+Known issues / Roadmap
+======================
+
+This moudle won't work cleanly if the field in question is present twice in the view.
```
Comment (bealdav) [inline company_dependent_flag/README.rst:45]:
```suggestion
This module won't work cleanly if the field in question is present twice in the view.
```

File: company_dependent_flag/readme/ROADMAP.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+This moudle won't work cleanly if the field in question is present twice in the view.
```
Comment (bealdav) [inline company_dependent_flag/readme/ROADMAP.rst:1]:
```suggestion
This module won't work cleanly if the field in question is present twice in the view.
```

**Review by bealdav — APPROVED** [review]
LGTM looks good except typo.

Needs to forward port in 16.0

Thanks a lot

**Comment by toita86** [comment]
@bealdav 
> Needs to forward port in 16.0

By checking on the runboat, the issue fixed by this PR is absent, Imo there is no need to increase the complexity of the module by doing a fwp.

**Comment by bealdav** [comment]
Nice to checked, thanks a lot @toita86 .

**Pull request merged**